exercises/sam_jenkins/ex02_oo_basics/src/Shapes.h

```diff
@@ -0,0 +1,24 @@
+//Base file for shapes to inherit from
+#ifndef GUARD_Shapes_h
+#define GUARD_Shapes_h
+#include <string>
+class Shapes {
+public:
+	
+	virtual double calcPerimiter() const =0;
+	virtual double calcArea() const =0;
+	//virtual std::string print() = 0;
+	std::string getType()const;
+	
+	int getSides()const;
```

Comment:
It might be an idea to `inline` this function so the compiler tries to optimise it.